.github/workflows/linter.yml

```diff
@@ -13,17 +13,22 @@ on:
     branches:         
     - master
 
+permissions:
+  contents: read
 
 jobs:
   yamllint:
+    permissions:
+      contents: read  # for actions/checkout to fetch code
+      pull-requests: write  # for karancode/yamllint-github-action to post comments on PRs
     name: Yamllint
     runs-on: ubuntu-latest
     steps:
 
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
 
     - name: 'Yamllint'
-      uses: karancode/yamllint-github-action@master
+      uses: karancode/yamllint-github-action@dd59165b84d90d37fc919c3c7dd84c7e37cd6bfb # master
```

Comment:
Is this effectively tip / there is no version?

Reply:
This is the most [recent commit](https://github.com/karancode/yamllint-github-action/commit/dd59165b84d90d37fc919c3c7dd84c7e37cd6bfb) into the master stream (dated April 2020), and is labeled as version 2.0.0. I'll update to the label for clarity.